### PR TITLE
fix(vue-instantsearch): prevent click on navigation items when pressing modifiers

### PR DESCRIPTION
**Pull request by sarahdayan**
This replicates the same behavior as in InstantSearch.js and React InstantSearch—whenever the end user is pressing a modifier key or the middle mouse button, we prevent from refining.

The behavior will be tested in https://github.com/algolia/instantsearch/pull/5836 which will be rebased onto it, so I skipped tests here to avoid doing the same work twice. Let me know if you think differently.

### CODE REVIEW

File: packages/vue-instantsearch/src/components/Pagination.vue [hunk 1]
```diff
@@ -135,14 +135,14 @@
             name="last"
             :createURL="() => state.createURL(state.nbPages - 1)"
             :is-last-page="state.isLastPage"
-            :refine="() => refine(state.nbPages - 1)"
+            :refine="() => refine(state.nbPages - 1, $event)"
```
Comment (Haroenv) [inline packages/vue-instantsearch/src/components/Pagination.vue:138]: does this work?

File: packages/vue-instantsearch/src/components/Pagination.vue [hunk 2]
```diff
@@ -107,14 +107,14 @@
...
-            :refine="() => refine(state.currentRefinement + 1)"
+            :refine="() => refine(state.currentRefinement + 1, $event)"
           >
             <template v-if="!state.isLastPage">
               <a
                 :class="suit('link')"
                 aria-label="Next"
                 :href="state.createURL(state.currentRefinement + 1)"
-                @click.prevent="refine(state.currentRefinement + 1)"
+                @click.prevent="refine(state.currentRefinement + 1, $event)"
```
Comment (Haroenv) [inline packages/vue-instantsearch/src/components/Pagination.vue:117]: can we use .exact? https://vuejs.org/guide/essentials/event-handling.html#key-modifiers

Comment (sarahdayan) [reply packages/vue-instantsearch/src/components/Pagination.vue:117]: Yes, it's actually much better! I forgot about this Vue magic.

**Review by Haroenv — APPROVED** [review]

File: packages/vue-instantsearch/src/components/Pagination.vue (hunk 1 above)
Comment (sarahdayan) [reply packages/vue-instantsearch/src/components/Pagination.vue]: That was actually unnecessary.

**Pull request merged**